pkgs/by-name/st/stdoutisatty/package.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  makeBinaryWrapper,
+  nix-update-script,
+  runCommand,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "stdoutisatty";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    owner = "lilydjwg";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    hash = "sha256-NyVn9cxx0rY1ridNDTqe0pzcVhdLVaPCKT4hoQkQzRs=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeBinaryWrapper
+  ];
+
+  preFixup = ''
+    # necessary for dynamic loading of `lib/libstdoutisatty.so` at runtime
+    wrapProgram $out/bin/${finalAttrs.pname} \
+      --prefix LD_LIBRARY_PATH : "$out/lib"
+  '';
```

Comment:
We should define `LIBDIR` instead: https://github.com/lilydjwg/stdoutisatty/blob/c77075065243f353a13c169941e14a6df7849038/stdoutisatty.c#L14, `LD_LIBRARY_PATH` is too big a hammer.

Reply:
Changed to use `cmakeFlags` in the newer version. In particular, since the upstream `CMakeLists.txt` specified `LIB_FILE`, we need to override `LIB_FILE` (instead of `LIBDIR`) for it to really work; see the logic around https://github.com/lilydjwg/stdoutisatty/blob/c77075065243f353a13c169941e14a6df7849038/stdoutisatty.c#L9-L16.